search-books/src/main/java/nix/education/service/TransactionalDataService.java

```diff
@@ -0,0 +1,38 @@
+package nix.education.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import nix.education.entity.Book;
+import nix.education.entity.Tag;
+import nix.education.repository.BookRepository;
+import nix.education.repository.TagRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TransactionalDataService {
```

Comment:
This class makes sense to me, when all the public method do some interaction with DB. Method `getTagTitles()` does nothing with DB. We need to think how we can reflect our intentions in the code in more elegant way

Reply:
I see. I'll try to come up with a new appropriate name for this class.